management/virtualmachine/client.go

```diff
@@ -19,23 +22,42 @@ const (
 	errParamNotSpecified = "Parameter %s is not specified."
 )
 
-//NewClient is used to instantiate a new VmClient from an Azure client
+//NewClient is used to instantiate a new VirtualMachineClient from an Azure client
 func NewClient(client management.Client) VirtualMachineClient {
 	return VirtualMachineClient{client: client}
 }
 
-func (self VirtualMachineClient) CreateDeployment(role Role, cloudServiceName string) (management.OperationId, error) {
-	data, err := xml.Marshal(DeploymentRequest{
+// CreateDeployment creates a deployment and then creates a virtual machine
+// in the deployment based on the specified configuration.
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157194.aspx
+func (vm VirtualMachineClient) CreateDeployment(
+	role Role,
+	cloudServiceName string,
+	options *DeploymentOptions) (management.OperationId, error) {
```

Comment:
My reasoning is that it's very user unfriendly (at least IMO) that if you don't need the extra options, you still need to create the object. Which is this case also has a pretty long name `virtualmachine.CreateDeploymentOptions{}`, so means some extra typing even through you don't use it.